### PR TITLE
Integrate with S3 and visually show values

### PR DESCRIPTION
**Pull request by stevenking86**
- Integrate the data from S3
- Display the data as a chart
- Display user data in <span> tags
- Some cleanup

TODO:
- Refactor my dumb 'set' methods in vals-chart.js 
- make chart re-render every so often
- figure out why we get errors on some records of our s3 records but not others. 
- fix s3Files waterfall js bs.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -5,31 +5,53 @@
...
+
     var avgCounter = 0;
     var volumes = [];
     var numLights = 0;
     var hueIp = null;
 
     setInterval(function() {
         hueIp = document.getElementById("hueIP").value;
-
+        updateLights(Mic.soundValues());
```
Comment (lnagle) [inline index.html]: I think this is extraneous. See same call below.

File: index.html [hunk 2]
```diff
@@ -5,31 +5,53 @@
...
+        </div>
+      </div>
+    </div>
 
-  <body>
     <script>
     var Mic = new Microphone();
 
+    allData = []
+    getS3Files();
```
Comment (lnagle) [inline index.html:45]: It's not immediately apparent how/when allData gets populated. It'd be preferable to have getS3Files take a callback that then sets allData (or pushes into allData if you'd rather not overwrite what's already there.)

```
getS3Files(function(s3Data) {
  allData = s3Data;
});
```

allData is also a bit ambiguous. `s3Data` may be more apropos.

File: aws.js [hunk 3]
```diff
@@ -39,7 +36,48 @@ function sendKinesisData(vals) {
...
+        Key: key
+      }
+
+      s3.getObject(params, function (err, data) {
+        if (err) {
+          console.log(err);
+        } else {
+          var splitString = data.Body.toString().split('}')
+          splitString.map(function(obj) {
+            allData.push(JSON.parse(obj + '}'))
```
Comment (lnagle) [inline aws.js:73]: RE: the comment about having a callback here:

```
var data = splitString.map(function(obj) {
  return JSON.parse(obj + '}');
}

callback(data);

File: aws.js [hunk 4]
```diff
@@ -39,7 +36,48 @@ function sendKinesisData(vals) {
...
+  var data = {}
+
+  s3.listObjects(params, function (err, data) {
+    if(err)throw err;
+
+    var keys = data.Contents.map(function(content) {
+      return content.Key;
+    });
+
+    keys.map(function(key) {
```
Comment (lnagle) [inline aws.js:61]: Rather than mapping again, this could probably be moved into line 58.

File: vals-chart.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+  highs = highs.filter(function( element ) {
+    return element !== undefined;
+  });
+
+  var highsSum = highs.reduce(function(a, b) { return a + b; });
+  var avg = highsSum / highs.length;
+
+  return Math.floor(avg)
+}
```
Comment (lnagle) [inline vals-chart.js]: We can DRY the above four functions with this.

```
/**
 * @param {string} soundCategory A string with one of these values: "vol", "bass", "mids", "highs".
 */
function setSoundCategory(soundCategory) {
  var data = allData.map(function(obj) {
    return obj[soundCategory];
  });

  data = data.filter(function(element) {
    return element !== undefined;
  });

  var sum = data.reduce(function(a, b) { return a + b });
  var avg = sum / soundCategory.length;

  return Math.floor(avg);
}

File: index.html (hunk 2 above)
Comment (stevenking86) [reply index.html:45]: I definitely should do this.

**Pull request merged**